gdk4/src/texture.rs

```diff
@@ -18,3 +21,51 @@ impl<O: IsA<Texture>> TextureExtManual for O {
         }
     }
 }
+
+impl Texture {
+    pub fn for_pixbuf_future(
+        pixbuf: &'static gdk_pixbuf::Pixbuf,
+    ) -> Pin<Box<dyn Future<Output = Option<Texture>> + 'static>> {
+        Box::pin(async move {
+            let (sender, receiver) = futures_channel::oneshot::channel();
+
+            let sender = std::cell::Cell::new(Some(sender));
+
+            let task = gio::Task::new(
+                Some(&pixbuf.upcast_ref()),
+                gio::Cancellable::NONE,
+                move |result, _| {
+                    let texture = result
+                        .downcast_ref::<gio::Task>()
+                        .unwrap()
+                        .propagate_value()
+                        .unwrap()
+                        .get::<Texture>()
+                        .unwrap();
+                    if let Some(m) = sender.replace(None) {
+                        let _result = m.send(texture);
+                    }
+                },
+            );
+
+            let task_func = move |task: &gio::Task,
+                                  source_object: Option<&glib::Object>,
+                                  _: Option<&gio::Cancellable>| {
+                let pixbuf = source_object
+                    .unwrap()
+                    .downcast_ref::<gdk_pixbuf::Pixbuf>()
+                    .unwrap();
+                let texture = Texture::for_pixbuf(&pixbuf);
+                task.return_value(&texture.to_value());
+            };
+
+            task.run_in_thread(task_func);
+
+            if let Ok(texture) = receiver.await {
```

Comment:
This should probably be a `Result` that is returned here

Reply:
I am not sure whats better, the original api does not return a Result, (but a result is used in the other functions that I plan to implement here).

Reply:
The original API should also return a result (and you can configure it in `Gir.toml` like that)

Reply:
The original api cannot fail, if for some reason the operation fails it will instead do a g_error. I am not sure how to handle that given that it will happen in another thread.

Reply:
`g_error()` immediately kills the whole process, not just a thread :) There's no way you can handle that.